File: 0086-partition-list/0086-partition-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def partition(self, head: Optional[ListNode], x: int) -> Optional[ListNode]:
        if(head==None or head.next==None):
            return head
        
        newHead = ListNode()
        p = newHead

        curr = head
        while(curr!=None):
            if(curr.val<x):
                p.next = ListNode(curr.val)
                p = p.next
            curr = curr.next
        
        curr = head
        while(curr!=None):
            if(curr.val>=x):
                p.next = ListNode(curr.val)
                p = p.next
            curr = curr.next
        
        return newHead.next

File: 0086-partition-list/0086-partition-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode partition(ListNode head, int x) {
        ListNode newHead = new ListNode();
        ListNode p = newHead;

        ListNode curr = head;
        while(curr!=null){
            if(curr.val<x){
                p.next = new ListNode(curr.val);
                p = p.next;
            }
            curr = curr.next;
        }

        curr = head;
        while(curr!=null){
            if(curr.val>=x){
                p.next = new ListNode(curr.val);
                p = p.next;
            }
            curr = curr.next;
        }

        return newHead.next;        
    }
}